industry_benchmarks/utils/data_gathering.py

```diff
@@ -444,14 +445,224 @@ def gather_ligand_scores(
 
     return all_ligand_scores
 
+def load_results_file(file_name: pathlib.Path) -> None | dict:
+    """Try and load the JSON file as a results file and check that the cleanup script has been used.
+
+    Raises
+    ------
+        ValueError: If the results clean up script has not been run.
+    """
+    with open(file_name, "r") as f:
+        results = json.load(f)
+
+    # First we check if someone passed in a network_setup.json
+    if results.get("__qualname__") == "AlchemicalNetwork":
+        print(f"{file_name} is a network_setup.json, skipping")
+        return None
+
+    #  Now  we check if someome passed in an input json
+    if results.get("__qualname__") == "Transformation":
+        print(f"{file_name} is an input json, skipping")
+        return None
+
+    # Check to see if we have already cleaned  up this result
+    result_key = next(k for k in results["protocol_result"]["data"].keys())
+    if (
+            "structural_analysis"
+            in results["protocol_result"]["data"][result_key][0]["outputs"]
+    ):
+        raise ValueError(f"{file_name} has not been clean up please make sure you run the `results_cleanup.py` script first.")
+
+    # Check to make sure we don't have more than one proto result
+    # We might have ProtocolUnitResult-* and ProtocolUnitFailure-*
+    # We only handle the case where we have one ProtocolUnitResult
+    protocol_unit_result_count = len(
+        [
+            k
+            for k in results["unit_results"].keys()
+            if k.startswith("ProtocolUnitResult")
+        ]
+    )
+
+    # Check to make sure we don't just have failures
+    # if all failures, tell user to re-run
+    if protocol_unit_result_count == 0:
+        print(f"{file_name} failed to run, traceback and exception below, this will not be included in the results. \n")
+        proto_failures = [
+            k
+            for k in results["unit_results"].keys()
+            if k.startswith("ProtocolUnitFailure")
+        ]
+        for proto_failure in proto_failures:
+            print("\n")
+            print(results["unit_results"][proto_failure]["traceback"])
+            print(results["unit_results"][proto_failure]["exception"])
+            print("\n")
+        return None
+
+    return results
+
+def remove_first_reversed_sequential_duplicate_from_path(path: pathlib.Path) -> pathlib.Path:
+    """
+    Remove the first duplicated directory from path
+    We reverse the path so we remove the first sequential
+    duplicated directory starting from the deepest part of the path
+    """
+
+    # reverse the path parts so we start from the deepest part first
+    reversed_path_parts = list(reversed(path.parts))
+    max_idx = len(reversed_path_parts)
+
+    # find index of first dupe
+    for idx in range(max_idx - 1):
+        if reversed_path_parts[idx] == reversed_path_parts[idx + 1]:
+            break
+    # no dupes
+    else:
```

Comment:
Should the `else` have the same indentation as `if`?

Reply:
Not for the `for else` pattern, this means only run the `else` block if the `for` loop does not trigger the break